### ISSUE TITLE
Calcite Buttons do not behave consistently with HTML Buttons within a CSS Flex container

### ISSUE THREAD

**Issue by gavinr-maps** [Esri/calcite-design-system#9025]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

When a calcite button is within a container that has `display: flex` set, it :
1. does not get its width set correctly
2. does not have the text wrap correctly

Here is the demo: https://codepen.io/gavinr/pen/YzMrEpX?editors=1000

### Expected Behavior

Calcite buttons should behave similar to plain HTML buttons when within a container that has `display: flex` set.

### Reproduction Sample

https://codepen.io/gavinr/pen/YzMrEpX?editors=1000

### Reproduction Steps

1. open https://codepen.io/gavinr/pen/YzMrEpX?editors=1000
2. See the top part, shows how HTML buttons work
3. See the bottom part - same CSS class applied to Calcite buttons but they are not behaving the same way as the top section
![image](https://github.com/Esri/calcite-design-system/assets/137905994/5c08a919-54cd-4612-9e18-e6f0397d7a4f)

### Reproduction Version

2.7.0

### Relevant Info

Latest Chrome

### Priority impact

p2 - want for current milestone

### Impact

I have created a workaround using JavaScript but it is using `.shadowRoot` which is not very preferable: https://codepen.io/gavinr/pen/zYXPEPd?editors=0010

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Professional Services - Midwest Delivery Center

**Comment by SkyeSeitz** [comment]
Agreed. Support for wrapping on `calcite-button` should be opt-in only to avoid oversized buttons when possible or mismatched button heights.

**Comment by ashetland** [comment]
> It also seems like the truncation behavior doesn’t work in the above scenario OOTB, so adjustments to that behavior can be looked at.

Using the `width` prop on the buttons also doesn't seem to work in the above example.
